src/Support/TypeToSchemaExtensions/ResponderTypeToSchema.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Dedoc\Scramble\Support\TypeToSchemaExtensions;
+
+use Dedoc\Scramble\Extensions\TypeToSchemaExtension;
+use Dedoc\Scramble\Support\Generator\Reference;
+use Dedoc\Scramble\Support\Generator\Response;
+use Dedoc\Scramble\Support\Generator\Schema;
+use Dedoc\Scramble\Support\Generator\Types as OpenApiTypes;
+use Dedoc\Scramble\Support\Generator\Types\UnknownType;
+use Dedoc\Scramble\Support\InferExtensions\ResourceCollectionTypeInfer;
+use Dedoc\Scramble\Support\Type\ArrayItemType_;
+use Dedoc\Scramble\Support\Type\ArrayType;
+use Dedoc\Scramble\Support\Type\Generic;
+use Dedoc\Scramble\Support\Type\Literal\LiteralBooleanType;
+use Dedoc\Scramble\Support\Type\ObjectType;
+use Dedoc\Scramble\Support\Type\Type;
+use Dedoc\Scramble\Support\Type\TypeWalker;
+use Dedoc\Scramble\Support\Type\Union;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Http\Resources\Json\AnonymousResourceCollection;
+use Illuminate\Http\Resources\Json\JsonResource;
+use Illuminate\Http\Resources\Json\ResourceCollection;
+use Illuminate\Http\Resources\MergeValue;
+use Illuminate\Http\Resources\MissingValue;
+use League\Fractal\TransformerAbstract;
+use Orchestra\Canvas\Commands\Database\Eloquent;
+
+class ResponderTypeToSchema extends TypeToSchemaExtension
+{
+    public function shouldHandle(Type $type)
+    {
+        return $type->isInstanceOf(TransformerAbstract::class);
+    }
+
+    /**
+     * @param  ObjectType  $type
+     */
+    public function toSchema(Type $type)
+    {
+        /*if ($type->isInstanceOf(Collection::class)) {
+            $array = (new ResourceCollectionTypeInfer)->getBasicCollectionType($definition);
+        } else {
+            return new UnknownType();
+        }*/
+
+        $array = ($def = $type->getMethodDefinition('transform'))
+            ? $def->type->getReturnType()
+            : new \Dedoc\Scramble\Support\Type\UnknownType();
+
+        if (!$array instanceof ArrayType) {
+            return new UnknownType();
+        }
+
+        $array->items = $this->flattenMergeValues($array->items);
+
+        return $this->openApiTransformer->transform($array);
+    }
+
+    private function flattenMergeValues(array $items)
```

Comment:
```suggestion
    /**
     * @param array<int, ArrayItemType_> $items
     *
     * @return array<ArrayItemType_>
     */
    private function flattenMergeValues(array $items)
```
I'm not sure if `ArrayItemType_` is the only type of data the array contains.